### PR TITLE
chore: simplify json cmake download package

### LINKED ISSUE #194: cmake compile always stop at `Detecting C compile features - done`
### Description
cmake compile always stop at `Detecting C compile features - done`

### Your System Info
ubuntu 18.04 
cmake 3.23.0-rc3
gcc 11.2.0

### Screenshots/Testimonials
image

### PR DESCRIPTION
**Pull request by sheny1xuan**
Signed-off-by: stonex <1479765922@qq.com>

## Fixes #194 
+ simplify json cmake download package

### CODE REVIEW

**Comment by hsluoyz** [comment]
@sheny1xuan plz fix:

image

**Comment by hsluoyz** [comment]
@EmperorYP7 plz review

**Comment by EmperorYP7** [comment]
> @sheny1xuan plz fix:
> image

This needs to be fixed in a separate PR. We need to track and fix the memory leakage throughout the app.

File: cmake/modules/Findjson.cmake [hunk 1]
```diff
@@ -14,16 +14,12 @@
 #  limitations under the License.
 include(FetchContent)
 
+set(JSON_Install ON)
+
 FetchContent_Declare(json
-  GIT_REPOSITORY https://github.com/nlohmann/json.git
-  GIT_TAG v3.10.1)
+  GIT_REPOSITORY https://github.com/ArthurSonzogni/nlohmann_json_cmake_fetchcontent
```
Comment (EmperorYP7) [inline cmake/modules/Findjson.cmake:20]:
```suggestion
  GIT_REPOSITORY https://github.com/ArthurSonzogni/nlohmann_json_cmake_fetchcontent.git
```

**Review by EmperorYP7 — CHANGES_REQUESTED** [review]

File: cmake/modules/Findjson.cmake (hunk 1 above)
Comment (sheny1xuan) [reply cmake/modules/Findjson.cmake:20]: Maybe it's not necessary to add suffix for this link. The official user manual don't have `.git`, and it works well in may local machine and ci.

**Review by EmperorYP7 — APPROVED** [review]

**Pull request merged**